### PR TITLE
feat(snowflake): implement insert method

### LINKED ISSUE #7955: bug: Snowflake insert is not working even for a very simple table
### What happened?

I have following ibis code for snowflake to reproduce the issue
```
import pandas
import ib is
df = pandas.DataFrame.from_dict({"name": ['n1']})
ibis_con = ibis.snowflake.connect(...)
ibis_con.create_table("TEST", schema=ibis.schema([("name", "string")]), overwrite=True)
ibis_con.insert('TEST', df) --> throw exception sqlalchemy.exc.ProgrammingError: (snowflake.connector.errors.ProgrammingError) 001003 (42000): SQL compilation error

# create table with the same df works
ibis_con.create_table("TEST", df, overwrite=True)
```

### What version of ibis are you using?

7.2.0

### What backend(s) are you using, if any?

snowflake

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by cpcloud**
Implements the `insert` method for snowflake. Closes #7955.

### CODE REVIEW

File: ibis/backends/snowflake/__init__.py [hunk 1]
```diff
@@ -973,6 +973,45 @@ def read_parquet(
...
+
+        self._run_pre_execute_hooks(obj)
+        query = sg.exp.insert(
+            expression=self.compile(obj),
+            into=sg.table(table_name, db=schema, catalog=database, quoted=True),
+            columns=[sg.column(col, quoted=True) for col in obj.columns],
+            dialect=self.name,
+        )
+        with self.begin() as con:
+            if overwrite:
```
Comment (cpcloud) [inline ibis/backends/snowflake/__init__.py:1010]: We can't pass this to `insert` because of https://github.com/tobymao/sqlglot/issues/2815

**Review by jcrist — APPROVED** [review]

**Comment by cpcloud** [comment]
Whoops this wasn't ready 😬

**Comment by cpcloud** [comment]
I'll pr another one

**Comment by jcrist** [comment]
oop, sorry, figured it was since the review request :grimacing:

**Pull request merged**